src/hazelcore/dynamics/Evaluator.re

```diff
@@ -20,13 +20,15 @@ let grounded_List = NotGroundOrHole(List(Hole));
 
 let ground_cases_of = (ty: HTyp.t): ground_cases =>
   switch (ty) {
-  | Hole => Hole
+  | Hole
+  | TyVarHole(_, _) => Hole
   | Bool
   | Int
   | Float
   | Arrow(Hole, Hole)
   | Sum(Hole, Hole)
-  | List(Hole) => Ground
+  | List(Hole)
+  | TyVar(_, _) => Ground
```

Comment:
TyVar shouldn't be ground, it depends on what it stands for. leave it as a failwith("Unimplemented") for now, since we don't have bound type variables yet.